com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/RenameApplicationsStep.java

```diff
@@ -0,0 +1,156 @@
+package com.sap.cloud.lm.sl.cf.process.steps;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import com.sap.cloud.lm.sl.cf.core.model.BlueGreenApplicationNameSuffix;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaApplication;
+import com.sap.cloud.lm.sl.cf.process.util.ApplicationProductizationStateUpdater;
+import com.sap.cloud.lm.sl.cf.process.util.ApplicationProductizationStateUpdaterBasedOnAge;
+import com.sap.cloud.lm.sl.cf.process.util.ApplicationProductizationStateUpdaterBasedOnColor;
+import org.cloudfoundry.client.lib.CloudControllerClient;
+import org.flowable.engine.delegate.DelegateExecution;
+import org.flowable.variable.api.delegate.VariableScope;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Scope;
+
+import com.sap.cloud.lm.sl.cf.core.helpers.ApplicationNameSuffixAppender;
+import com.sap.cloud.lm.sl.cf.core.model.ApplicationColor;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMta;
+import com.sap.cloud.lm.sl.cf.process.helpers.ApplicationColorDetector;
+import com.sap.cloud.lm.sl.cf.process.message.Messages;
+import com.sap.cloud.lm.sl.common.ConflictException;
+import com.sap.cloud.lm.sl.mta.model.DeploymentDescriptor;
+
+import java.util.List;
+
+@Named("renameApplicationsStep")
+@Scope(BeanDefinition.SCOPE_PROTOTYPE)
+public class RenameApplicationsStep extends SyncFlowableStep {
+
+    @Inject
+    private ApplicationColorDetector applicationColorDetector;
+
+    @Override
+    protected StepPhase executeStep(ExecutionWrapper execution) {
+        RenameFlow renameFlow = createFlow(execution.getContext());
+        renameFlow.execute(execution);
+        return StepPhase.DONE;
+    }
+
+    private RenameFlow createFlow(VariableScope context) {
+        if (StepsUtil.getKeepOriginalAppNamesAfterDeploy(context)) {
+            return new AddOldNewSuffix();
```

Comment:
This step is named 'RenameApplicationsStep' -> it is logically the Flows to be named with the same pattern.

Reply:
Done